### DISCUSSION TITLE
Berry, how to access knx functions of tasmota?

### DISCUSSION THREAD

**Discussion by suppeee** [arendst/Tasmota#20584] [Q&A] [answered]
Hi there,

i'd like to access the knx-functions of tasmota via berry scripting. Is there a way (maybe still undocumented?) or is it worked on? i would need some elements whithin berry which can generate and send knx-ip-packets (udp-multicast). Also a "server" which waits for packets and parses the content (Group-Adresses, Values etc.) for rules.

Another idea would be to create individual knx-ip-packets with the udp class. However the logic is already somewhere in tasmota, so why not reuse it? 

My usecase is an ewelink zigbee bridge, which should "route" packets/information between zigbee and the knx-bus to enable interaction without the need for specific homeserver logic.

Thanks
johnny

**Comment by sfromis** [comment, marked as answer]
As there currently is no Berry/KNX integration, the only thing you have available now (outside of doing it all yourself) is using `tasmota.cmd` along with the regular [KNX commands](https://tasmota.github.io/docs/Commands/#knx) including sending of commands/values, and with parsing of the responses returned. You can also pickup any messages being published as JSON payloads, using `tasmota.add_rule` etc.

**Comment by Noschvie** [comment]
@suppeee Are you using KNX with Berry and if so, do you have an example? Thanks!

**Comment by suppeee** [comment]
sorry, no - right now i'm using a berry script to send the zigbee messages
via tcp to a homeserver (which is connected to knx) for the one way and
<ip>/cm?cmnd=zbsend for the way back. although it's quite a mess it works
very stable.
